features/cellular/TESTS/api/cellular_sms/main.cpp

```diff
@@ -82,41 +108,27 @@ static void init()
     TEST_ASSERT(at_init->get_last_error() == NSAPI_ERROR_OK);
 
     delete at_init;
-
-#if defined (MDMRTS) && defined (MDMCTS)
-    cellular_serial.set_flow_control(SerialBase::RTSCTS, MDMRTS, MDMCTS);
-#endif
-    cellularConnectionFSM.set_serial(&cellular_serial);
-    cellularConnectionFSM.set_callback(&cellular_status);
-
-    TEST_ASSERT(cellularConnectionFSM.init() == NSAPI_ERROR_OK);
-    TEST_ASSERT(cellularConnectionFSM.start_dispatch() == NSAPI_ERROR_OK);
-
-    
-    CellularDevice *device = cellularConnectionFSM.get_device();
-
-    TEST_ASSERT(device != NULL);
-    device->set_timeout(30000);
-
-    sms = device->open_sms(&cellular_serial);
-    TEST_ASSERT(sms != NULL);
-
-    wait(3);
-    
 }
 
-static void activate_context()
+static void init()
 {
-    CellularNetwork *network = cellularConnectionFSM.get_network();
+    // First store current service address
+    store_service_center_address();
+
+    createFSM();
```

Comment:
Why is createFSM call needed here, if it is also called in store_service_center_address?

Reply:
FSM is also deleted in store_service_center_address so that we can create unique ATHandler as we can't access it from public API's.